### PR TITLE
Removed dead AI code

### PR DESCRIPTION
**Pull request by reaperrr**
Edit: Changed PR to remove unused code instead.

### CODE REVIEW

**Comment by reaperrr** [comment]
Note: It is hard to tell how much this effectively helps compared to bleed, so this will require some more extensive testing.

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 1]
```diff
@@ -98,6 +98,12 @@ public class BuildingCategories
 		[Desc("Delay (in ticks) until rechecking for new BaseProviders.")]
 		public readonly int CheckForNewBasesDelay = 1500;
 
+		[Desc("Delay (in ticks) before deciding which enemy to attack first.")]
+		public readonly int ChooseInitialEnemyDelay = 750;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: Can you elaborate a bit why you added this?
I mean, couldn't we just use `UpdateAggroInterva` only instead?

Comment (reaperrr) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: I just implemented the intitial delay first and added the interval later. You're right, it's better to keep just the interval.

**Comment by reaperrr** [comment]
Updated.

**Comment by reaperrr** [comment]
Updated, changed `*Interval` to `*Delay` to avoid an upgrade rule later. As @GraionDilach mentioned in #9810, we need to rename Interval to Delay for full consistency.

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 2]
```diff
@@ -559,6 +565,36 @@ public void Tick(Actor self)
...
+
+			var leastLikedEnemies = liveEnemies
+				.GroupBy(e => aggro[e].Aggro)
+				.MaxByOrDefault(g => g.Key);
+
+			var enemy = (leastLikedEnemies != null) ?
+				leastLikedEnemies.Random(Random) : liveEnemies.FirstOrDefault();
+
+			// Bump the aggro slightly to avoid changing our mind
+			if (leastLikedEnemies.Count() > 1)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: What happens when `leastLikedEnemies` is `null`?

Comment (reaperrr) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: I'm not sure whether there are any circumstances under which `leastLikedEnemies` can be `null` as long as there are still `liveEnemies`. I assumed there aren't any, but I'm not 100% sure how `MaxByOrDefault` works.

But yes, if it's possible for `leastLikedEnemies` to be null, that case needs to be covered, and if not, the `var enemy` can be simplified.

**Comment by reaperrr** [comment]
Updated:
- now covers the case when `leastLikedEnemies` is `null` but `enemy` is not
- moved `var deadEnemies` behind the `liveEnemies.Any()` check so it's only performed if there are any live enemies remaining

**Comment by RoosterDragon** [comment]
So a quick breeze over HackyAI shows that `Aggro` is only used within `ChooseEnemyTarget` and `ChooseEnemyTarget` is unused!

Everything else seems to find `FindClosestEnemy` to decide what to attack, which doesn't care about aggro.

Looks like some refactoring may be in order!

**Comment by pchote** [comment]
Well that would explain why the AI suddenly turned dumb a couple of releases ago.

**Comment by reaperrr** [comment]
So the AI sittling idle sometimes is due to the closest enemy being too far away...

**Comment by RoosterDragon** [comment]
There is no upper limit to the search radius for finding an enemy most of the time, so at a glance I would expect the AI to attack things across the map if it had to.

This will probably need some deeper investigation to determine why the AI idles.

**Comment by GraionDilach** [comment]
I've seen the AI going against worms in D2k immediately instead of attacking each other - during reproducing that activity-stack-overflow report - so I'd guess the AI ends up aggroing the wrong player. Couldn't check it yet how this PR helps on that though.

**Comment by RoosterDragon** [comment]
First bad commit is e0b7242f1b0dac2c282aadbb2ecbca5fb6f3d1e1 (#2736, #2600), which would've affected release-20130514.

Given we've been without aggro for so long, it may be more prudent to remove the dead code and fix the existing code as needed.

**Comment by reaperrr** [comment]
> So a quick breeze over HackyAI shows that Aggro is only used within ChooseEnemyTarget and ChooseEnemyTarget is unused!

It's used when an actor is damaged, Aggro against the attack's owner is buffed by the damage in that case (unless that's somehow not working either, but my impression is that it works, the AI usually responds to being attacked).

Anyway, AI is not exactly my strong point so I changed this PR to just remove the dead `ChooseEnemyTarget()` code for now.

**Comment by RoosterDragon** [comment]
Sure, being attacked updates the aggro value. But nothing ever reads that value and uses it! So it has no effect in the end. You may as well get rid of it entirely.

(Also, you should remove the "fixes" in the OP lest merging this closes those now)

**Comment by obrakmann** [comment]
:+1:

**Comment by reaperrr** [comment]
Updated, removed the remaining unused aggro left-overs.

**Comment by pchote** [comment]
It seems a shame to remove this instead of fixing it, but removing is better than leaving it broken so :+1:

**Comment by pchote** [comment]
(no changelog because this has no player- or modder-visible changes)

**Pull request merged**